src/main/java/CliUI.java

```diff
@@ -1,87 +1,153 @@
             case 2: createSnippet(); break;
             case 3: deleteSnippet(); break;
             case 4: editSnippet(); break;
             case 5: filterSnippets(); break;
-            case 6: isOpen = false; break;
+            case 6: searchSnippets(); break;
+            case 7: isOpen = false; break;
             default: System.out.println(input + ": please select a valid option");
         }
     }
 
     private void createSnippet() {
         System.out.println("Enter the title of the snippet:");
-        String title = keyboard.next();
+        String title = keyboard.nextLine();
 
         System.out.println("Enter the language of the snippet:");
-        String lang = keyboard.next();
+        String lang = keyboard.nextLine();
 
         System.out.println("Enter the tags of the snippet, like tag1,tag2,...:");
-        String[] tags = keyboard.next().split(",");
+        String[] tags = keyboard.nextLine().split(",");
 
-        Integer snippetId = snippetManager.create(title, "", lang, tags);
-        snippetManager.edit(snippetId);
+        snippetManager.create(title, null, lang, tags);
     }
 
     private void deleteSnippet() {
         System.out.println("Enter the id of the snippet to delete:");
-        int input = keyboard.nextInt();
+        int input = getIntInput();
         snippetManager.delete(input);
     }
 
     private void editSnippet() {
         System.out.println("Enter the id of the snippet to edit");
-        Integer snippetId = keyboard.nextInt();
+        Integer snippetId = getIntInput();
         snippetManager.edit(snippetId);
     }
 
     private void filterSnippets() {
-        System.out.println("Enter the title to filter by (... for none):");
-        String title = keyboard.next();
-        if (title.equals("...")) {
-            title = "";
+        System.out.println("Enter the term to filter by (... for none):");
+        String filterTerm = keyboard.nextLine();
+        if (filterTerm.equals("...")) {
+            filterTerm = "";
         }
 
         System.out.println("Enter the language of the snippet (... for none):");
-        String lang = keyboard.next();
+        String lang = keyboard.nextLine();
         if (lang.equals("...")) {
             lang = "";
         }
 
         System.out.println("Enter the tags of the snippet, like tag1,tag2,...: (... for none)");
-        String[] tags = keyboard.next().split(",");
+        String[] tags = keyboard.nextLine().split(",");
         if (tags[0].equals("...")) {
             tags[0] = "";
         }
 
-        System.out.println(snippetManager.listSnippets(snippetManager.filter(title, lang, tags)));
+        System.out.println(snippetManager.listSnippets(snippetManager.filterSnippets(filterTerm, lang, tags)));
+    }
+
+    private void searchSnippets() {
+        System.out.println("Enter the term(s) to search by (... for none):");
+        String searchTerm = keyboard.nextLine();
+        if (searchTerm.equals("...")) {
```

Comment:
I agree with the others that it would be better to replace the "..." with something else.